SurfaceTopography/IO/FromFile.py

```diff
@@ -315,12 +315,14 @@ def read_block(fobj):
     # Height are in nm, width in mm
     if physical_sizes is None:
         physical_sizes = (nx * pixel_size, ny * pixel_size * aspect)
-    surface = Topography(np.fliplr(data), physical_sizes,
-                         info={**info, **dict(unit='mm')}, periodic=periodic)
     if height_scale_factor is None:
-        surface = surface.scale(wavelength / mult * 1e-6)
-    else:
-        surface = surface.scale(height_scale_factor)
+        height_scale_factor = wavelength / mult * 1e-6
+
+    surface = Topography(np.fliplr(data), physical_sizes,
+                         info={**info, **dict(unit='mm', height_scale_factor=height_scale_factor)},
+                         periodic=periodic)
+
+    surface = surface.scale(height_scale_factor)
```

Comment:
Or should we modify the "scale" method to include the height_scale_factor in the info dict ?

Reply:
Probably a good idea, this would avoid the problem for all file types. I would expect the key
in the dict as soon as `scale` is in the pipeline. What do you think @pastewka , could this cause any problems?

Reply:
Nevertheless the short term solution is probably to do this. I approve the PR.

Reply:
Well, we should think about it twice..if `scale` appears again in the pipeline, the resulting factor
in the info dict should be the product of all given scale factors, right? This should probably be implemented
by reviewing all `scale` arguments in the pipeline. This should be possible now using the `pipeline()` method, correct?

Reply:
The current way we handle this is not super ideal. I would not relay on the presence of the info dictionary in TopoBank, but check whether the returned topography is a scaled one. (It should have the `scale_factor` property... but take care, that property will be renamed to `heights_scale_factor` soon.)

Reply:
Do you mean `height_scale_factor without 's'? This is how the key in the info dict was called.

Ah ok, I've had only seen `._scale_factor` so far, with underscore, and therefore didn't use it.

So I'll use the `scale_factor` property then or check for `height_scale_factor` or `heights_scale_factor`.
Then this PR wouldn't be needed - although it makes things more consistent for now.

Reply:
But I thought that all surfaces are scaled - this wouldn't work then. The `scale_factor` comes from the `ScaledTopography` pipeline class.

Reply:
Yes, in TopoBank all topographies are scaled.

Well, now I see: Actually I need the information about the height scale factor **from the channel**, not 
from the topography. Currently when entering the information into the form in TopoBank, only the reader is loaded
and I don't have the topography yet. And the channels from the reader have no `scale_factor`. Currently they have the following properties:
```python
 'area_per_pt',
 'dim',
 'index',
 'info',
 'is_periodic',
 'name',
 'nb_grid_pts',
 'physical_sizes',
 'pixel_size',
```

So as alternative to the info dict,  we could we add the properties `height_scale_factor` to the channel, for all "wrapped readers" where the resulting topography already has `scale_factor` like for `read_opd`?
I think I could implement this here.

Reply:
Yes, that would make more sense. This is where it belongs, as the channels don't know about the pipeline system.

Reply:
Okay, thank you. I'm currently implementing this.

Here I stumbled over that about half of the file formats do nothing if a parameter `height_scale_factor` is given
to reader's `.topography()`. Is this intended?

What I would expect, e.g. if called with `.topography(..., height_scale_factor=2, ..)`, is that the original height scale factor from file is replaced with factor 2:
- e.g. if the original factor was 1.5, then the effective factor is 2/1.5 = 4/3.
- or if there is no height_scale_factor in file, I would expect that the resulting topography is scaled by factor 2 = 2/1.

Do you agree? 

Or is it correct that `height_scale_factor` is ignored in half of the cases? This would cause a problem in the logic of TopoBank.